src/include/miopen/image_transform/adjust_saturation/invoke_params.hpp

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include "miopen/invoke_params.hpp"
+#include "miopen/tensor.hpp"
+
+namespace miopen {
+namespace image_transform {
+namespace adjust_saturation {
+struct InvokeParams : public miopen::InvokeParams
+{
+    InvokeParams() = default;
+
+    const TensorDescriptor* inputTensorDesc;
+    const TensorDescriptor* outputTensorDesc;
+
+    ConstData_t input_buf = nullptr;
+    Data_t workspace_buf  = nullptr;
+    Data_t output_buf     = nullptr;
+
+    float saturation_factor = 0.0f;
+
+    size_t workspace_size = 0;
+
+    size_t GetWorkspaceSize() const { return workspace_size; }
+    Data_t GetWorkspace() const { return nullptr; }
```

Comment:
Maybe the general usage seems to be like this. 
```Data_t GetWorkspace() const { return workspace_buf; }```

Reply:
I've changed it to return the correct buffer